### PR TITLE
fix/violation-delete

### LINKED ISSUE #2539: Delete a violation config from the admin gui will not remove it from the class table.
When we delete a violation config from the admin gui, this config is still present in the class table and continue to trigger violation.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Removes violation from database when deleting violation from config

Impacts
=======
Violation handling in the config

Issue
=====
Fixes #2539

Delete branch after merge
=========================
NO

NEWS file entries
=================

Bug Fixes
------------
* Fixes violation definition sync when removing a violation from the config

### CODE REVIEW

**Comment by fdurand** [comment]
It fix an issue when you remove a violation config but also delete the current open violation for this violation id.
Do we have to archive instead of removing ?

**Comment by jrouzierinverse** [comment]
Remove class not in config

**Comment by jrouzierinverse** [comment]
- [x] Delete violation show message before deleting. 
- [x] Disable from the index page.

**Comment by jrouzierinverse** [comment]
Please review

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Violation.pm [hunk 1]
```diff
@@ -108,6 +109,29 @@ sub index :Path :Args(0) {
...
+    my ($status,$status_msg) = $model->update(
+        $item_id,
+        {enabled => $enabled},
+    );
+    $self->audit_current_action($c, status => $status);
+    $c->response->status($status);
+    $c->stash(
+        status_msg => $status_msg,
+        current_view => 'JSON',
+    ); # TODO: localize error message
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Violation.pm]: Todo

**Comment by jrouzierinverse** [comment]
Done

File: html/pfappserver/root/violation/list.tt [hunk 2]
```diff
@@ -9,7 +9,8 @@
           <h3><i>[% l('Delete Violation') %]</i> <span></span></h3>
         </div>
         <div class="modal-body">
-          <p>[% l('Really delete this violation?') %]</p>
+          <p>[% l('Deleting this violation will close all triggered violations associated with this violation.') %]</p>
```
Comment (fdurand) [inline html/pfappserver/root/violation/list.tt]: Deleting this violation will remove ...

**Review by fdurand — CHANGES_REQUESTED** [review]

**Comment by jrouzierinverse** [comment]
Done

**Pull request merged**